diff --git a/stm32mp157f-dk2.dts.orig b/stm32mp157f-dk2.dts
index 0748828..3286c35 100644
--- a/arch/arm/boot/dts/st/stm32mp157f-dk2.dts
+++ b/arch/arm/boot/dts/st/stm32mp157f-dk2.dts
@@ -161,3 +161,16 @@
 		vddio-supply = <&v3v3>;
 	};
 };
+
+&i2c5 {
+        pinctrl-names = "default", "sleep";
+        pinctrl-0 = <&i2c5_pins_a>;
+        pinctrl-1 = <&i2c5_sleep_pins_a>;
+        i2c-scl-rising-time-ns = <185>;
+        i2c-scl-falling-time-ns = <20>;
+        clock-frequency = <100000>;
+        /* spare dmas for other usage */
+        /delete-property/dmas;
+        /delete-property/dma-names;
+        status = "okay";
+};
